index.php

```diff
@@ -0,0 +1,147 @@
+            ],
+            'endOnSilence' => '3',
+            'endOnKey' => '#',
+            'beepOnStart' => true
+        ],
+        [
+            'action' => 'talk',
+            'text' => 'Thank you for your message. Goodbye.'
+        ],
+        [
+            'action' => 'notify',
+            'payload' => ['foo'=>'bar'],
+            'eventUrl' => [
+                $uri->getScheme().'://'.$uri->getHost().':'.$uri->getPort().'/webhooks/transcribe'
+            ],
+            'eventMethod' => "POST"
+        ],
+
+    ];
+
+    $response->getBody()->write(json_encode($ncco));
+    return $response
+        ->withHeader('Content-Type', 'application/json');
+});
+
+/**
+ * Event URL used by Nexmo for update POSTs
+ */
+$app->post('/webhooks/recording', function (Request $request, Response $response) {
+
+    $params = $request->getParsedBody();
+
+    error_log($params['recording_url']);
+
+    return $response
+        ->withStatus(204);
+});
+
+/**
+ * Fetch the voice recording from Nexmo, store on AWS S3
+ */
+$app->post('/webhooks/fetch', function (Request $request, Response $response) {
+
+    $params = json_decode($request->getBody(), true);
+
+    // Create Nexmo Client
+    $keypair = new Keypair(
+        file_get_contents($_ENV['NEXMO_APPLICATION_PRIVATE_KEY_PATH']),
+        $_ENV['NEXMO_APPLICATION_ID']
+    );
+
+    $nexmoClient = new Client($keypair);
+
+    $data = $nexmoClient->get($params['recording_url']);
+
+    // Create AWS S3 Client
+    $S3Client = new S3Client([
+        'credentials' => [
+            'key'    => $_ENV['AWS_ACCESS_KEY_ID'],
+            'secret' => $_ENV['AWS_SECRET_ACCESS_KEY'],
+        ],
+        'region' => $_ENV['AWS_REGION'],
+        'version' => 'latest',
+    ]);
+
+    $adapter = new AwsS3Adapter($S3Client, $_ENV['AWS_S3_BUCKET_NAME']);
+
+    $filesystem = new Filesystem($adapter);
+
+    $filesystem->put('/' . $_ENV['AWS_S3_RECORDING_FOLDER_NAME'] .'/'.$params['conversation_uuid'].'.mp3', $data->getBody());
+
+    return $response
+        ->withStatus(204);
+});
+
+/**
+ * Request Amazon Transcription of the voice file
+ */
+$app->post('/webhooks/transcribe', function (Request $request, Response $response) {
+
+    $params = json_decode($request->getBody(), true);
+
+    // Create Amazon Transcribe Client
+    $awsTranscribeClient = new TranscribeServiceClient([
+        'credentials' => [
+            'key'    => $_ENV['AWS_ACCESS_KEY_ID'],
+            'secret' => $_ENV['AWS_SECRET_ACCESS_KEY'],
+        ],
+        'region' => $_ENV['AWS_REGION'],
+        'version' => 'latest',
+    ]);
+
+    $transcriptionResult = $awsTranscribeClient->startTranscriptionJob([
+            'LanguageCode' => 'en-US',
+            'Media' => [
+                'MediaFileUri' => 'https://' . $_ENV['AWS_S3_BUCKET_NAME'] . '.s3.amazonaws.com/' . $_ENV['AWS_S3_RECORDING_FOLDER_NAME'] . '/' . $params['conversation_uuid'] . '.mp3',
```

Comment:
Where does `conversation_uuid` come from in the request? Is the user triggering this manually?

Reply:
It is being pulled from the body of the request via the notify NCCO object.